sw/device/tests/rv_core_ibex_rnd.c

```diff
@@ -0,0 +1,102 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_edn.h"
+#include "sw/device/lib/dif/dif_rv_core_ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "rv_core_ibex_regs.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+dif_result_t dif_rv_core_ibex_read_rnd_data_followed_by_status(
+    const dif_rv_core_ibex_t *rv_core_ibex, uint32_t *data,
+    dif_rv_core_ibex_rnd_status_t *followed_status) {
+  if (rv_core_ibex == NULL || data == NULL || followed_status == NULL) {
+    return kDifBadArg;
+  }
+  uint32_t reg = mmio_region_read32(rv_core_ibex->base_addr,
+                                    RV_CORE_IBEX_RND_STATUS_REG_OFFSET);
+  if (!bitfield_bit32_read(reg, RV_CORE_IBEX_RND_STATUS_RND_DATA_VALID_BIT)) {
+    return kDifError;
+  }
+
+  *data = mmio_region_read32(rv_core_ibex->base_addr,
+                             RV_CORE_IBEX_RND_DATA_REG_OFFSET);
+  *followed_status = mmio_region_read32(rv_core_ibex->base_addr,
+                                        RV_CORE_IBEX_RND_STATUS_REG_OFFSET);
+  return kDifOk;
+}
+
+dif_result_t dif_rv_core_ibex_multiple_unchecked_read_rnd_data(
+    const dif_rv_core_ibex_t *rv_core_ibex, const uint32_t repetitions,
+    uint32_t *data) {
+  if (rv_core_ibex == NULL || data == NULL) {
+    return kDifBadArg;
+  }
+
+  for (uint32_t i = 0; i < repetitions; i++) {
+    *data = mmio_region_read32(rv_core_ibex->base_addr,
+                               RV_CORE_IBEX_RND_DATA_REG_OFFSET);
+  }
+  return kDifOk;
+}
+
+#define RANDOM_DATA_READS 16
+
+bool test_main(void) {
+  // Verify the functionality of the random number generation CSRs.
+
+  // Add variables to the stack.
+  dif_rv_core_ibex_t rv_core_ibex;
+  uint32_t rnd_data, previous_rnd_data;
+  dif_rv_core_ibex_rnd_status_t rnd_status = 0;
+  dif_edn_t edn0;
+  dif_edn_seed_material_t seed_material;
+  dif_result_t return_value;
+  const uint32_t unchecked_read_repititions = 64;
```

Comment:
Const values should be named using camel case and starting with `k`.
```suggestion
  const uint32_t kUncheckedReadRepititions = 64;
```